torch/lib/THD/base/init_methods/InitMethodEnv.cpp

```diff
@@ -37,40 +37,51 @@ std::tuple<std::string, port_type> load_worker_env() {
   return std::make_tuple(must_getenv(MASTER_ADDR_ENV), port);
 }
 
-rank_type load_rank_env() {
-  return convertToRank(std::stol(must_getenv(RANK_ENV)));
+rank_type getRank(int rank) {
+  const char *env_rank_str = std::getenv(RANK_ENV);
+  int env_rank = rank;
+  if (env_rank_str != nullptr)
+    env_rank = std::stol(env_rank_str);
+  if (rank != -1 && env_rank != rank)
```

Comment:
This `if` is correct for this error but we miss different `if` which checks if any of this is set.